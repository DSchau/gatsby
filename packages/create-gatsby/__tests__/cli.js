jest.mock('gatsby-cli/lib/create-cli', () => jest.fn())
const createCli = require(`gatsby-cli/lib/create-cli`)
const createGatsbyCli = require(`../cli`)

beforeEach(() => {
  createCli.mockClear()
})

test(`it forwards arguments to gatsby-cli`, () => {
  const args = [`new`, `my-gatsby-app`, `--leeoroy`, `--jenkins`]

  createGatsbyCli(args)

  expect(createCli).toHaveBeenCalledWith(args)
})

test(`it throws if new command not specified`, () => {
  const args = [`gatsby-app`]

  expect(() => createGatsbyCli(args)).toThrowErrorMatchingSnapshot()
})
