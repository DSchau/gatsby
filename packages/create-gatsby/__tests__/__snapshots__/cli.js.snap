// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it throws if new command not specified 1`] = `
"create-gatsby can only be used with the new command, e.g. \`npm init gatsby new your-gatsby-app\`
See gatsby.app/cli for more info."
`;
